src/main/java/seedu/resireg/logic/CommandHistory.java

```diff
@@ -53,11 +69,12 @@ public boolean equals(Object o) {
 
         // state check
         CommandHistory other = (CommandHistory) o;
-        return history.equals(other.history);
+        return history.equals(other.history)
+                && counter.equals(other.counter);
```

Comment:
Lacks test coverage. (Probably add a test where there's two `CommandHistory` objects with the same `history` but different `counter`). Useful if for some reason the list of command histories gets modified unsafely.

Reply:
Resolved.